diff_drive_controller/test/diff_drive_nan_test.cpp

```diff
@@ -91,6 +91,16 @@ TEST_F(DiffDriveControllerTest, testNaNCmd) {
   for (int i = 0; i < 10; ++i) {
     geometry_msgs::Twist cmd_vel;
     cmd_vel.linear.x = NAN;
+    cmd_vel.angular.z = 0.0;
+    publish(cmd_vel);
+    geometry_msgs::TwistStamped odom_msg = getLastCmdVelOut();
+    EXPECT_FALSE(std::isnan(odom_msg.twist.linear.x));
+    EXPECT_FALSE(std::isnan(odom_msg.twist.angular.z));
+    ros::Duration(0.1).sleep();
+  }
+  for (int i = 0; i < 10; ++i) {
+    geometry_msgs::Twist cmd_vel;
+    cmd_vel.linear.x = 0.0;
```

Comment:
something i noticed when testing this fix for the test was that it still doesn't catch the error unless the other test in this file is commented. Anyone have any ideas about this? Is rostest reusing the same master and running the tests in parallel or something?

Reply:
As we discussed offline, the tests within a test suite are run sequentially on the same master. However, each `.test` is run with its own master.